src/globals.c

```diff
@@ -225,13 +248,21 @@ int set_content_url(char *url)
 int set_version_url(char *url)
```

Comment:
This function is starting to became very confusing. I think it's worth refactoring that and changing how its used so we don't need to have logic in it to decide when to overlap and when not to overlap

Reply:
The same for set_content_url()

Reply:
done